protos/sigstore_verification.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.verification.v1;
+
+import "sigstore_common.proto";
+import "sigstore_trustroot.proto";
+import "sigstore_bundle.proto";
+
+option go_package = "github.com/sigstore/proto/verification/v1";
+option java_package = "dev.sigstore.proto.verification.v1";
+option java_multiple_files = true;
+option java_outer_classname = "VerificationProto";
+
+// The identity of a signer.
+message Identity {
+        // The X.509v3 issuer extension (OID 1.3.6.1.4.1.57264.1.1)
+        string issuer = 1;
+        // A regular expression describing the expected value for the SAN.
+        string subject_alternative_name = 2;
+}
+
+message TransparencyLogs {
+        // The set of transparency logs to use during bundle verification.
+        repeated dev.sigstore.trustroot.v1.TransparencyLogInstance instances = 1;
+        // The minimim number of logs the entry must appear on.
+        // Setting this value to zero disables verification.
+        int32 threshold = 2;
```

Comment:
I was thinking of this too, it may be the that the message name `TransparencyLogs` is badly named. One option I thought about was to move the options i.e `threshold` to the `Options` object. If so, the entire message `TransparencyLogs` could be removed. I grouped them together and the idea is to express "expect at least `$threshold` numbers of inclusion promises of these transparency logs".

Reply:
I like this idea, but this does require some additional thought about how this is documented. I don't want it to seem like you must have a threshold > 1 in order to use sigstore.

Reply:
As some of the ideas added here are a bit beyond the current deployment model and what people may use, I can move them around to have an `Input` structure for a simplified structure with an optional advanced input.